From 9e2682ecd7ed17033e63e49f847a7d74a43f3f88 Mon Sep 17 00:00:00 2001
From: David King <amigadave@amigadave.com>
Date: Tue, 25 Aug 2015 10:28:34 +0100
Subject: [PATCH] tracker-miner-fs: avoid crash during init

Correctly handle the result of tracker_sparql_cursor_get_string(), which
can be NULL.

https://bugzilla.gnome.org/show_bug.cgi?id=754055
---
 src/miners/fs/tracker-miner-files.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/miners/fs/tracker-miner-files.c b/src/miners/fs/tracker-miner-files.c
index fb7445f..527c6e5 100644
--- a/src/miners/fs/tracker-miner-files.c
+++ b/src/miners/fs/tracker-miner-files.c
@@ -929,6 +929,9 @@ init_mount_points (TrackerMinerFiles *miner_files)
 
 		urn = tracker_sparql_cursor_get_string (cursor, 0, NULL);
 
+		if (!urn)
+			continue;
+
 		if (strcmp (urn, TRACKER_DATASOURCE_URN_NON_REMOVABLE_MEDIA) == 0) {
 			/* Report non-removable media to be mounted by HAL as well */
 			state |= VOLUME_MOUNTED;
-- 
2.5.0

